File: AndroidSDKCore/src/main/java/com/leanplum/utils/CTUtils.java
package com.leanplum.utils;

import android.annotation.SuppressLint;

import com.clevertap.android.sdk.CleverTapAPI;
import com.clevertap.android.sdk.CleverTapInstanceConfig;
import com.clevertap.android.sdk.LocalDataStore;
import com.clevertap.android.sdk.task.CTExecutorFactory;
import com.clevertap.android.sdk.task.CTExecutors;
import java.util.HashMap;
import java.util.Map;

public class CTUtils {

  @SuppressLint("RestrictedApi")
  public static void ensureLocalDataStoreValue(String key, CleverTapAPI cleverTapApi) {
    LocalDataStore localDataStore = cleverTapApi.getCoreState().getLocalDataStore();
    Object value = localDataStore.getProfileProperty(key);
    if (value == null) {
      Map<String, Object> map = new HashMap<>();
      map.put(key, "");
      localDataStore.updateProfileFields(map);
    }
  }

  @SuppressLint("RestrictedApi")
  public static void addMultiValueForKey(String key, String value, CleverTapAPI cleverTapApi) {
    CleverTapInstanceConfig config = cleverTapApi.getCoreState().getConfig();

    CTExecutors executors = CTExecutorFactory.executors(config);
    executors
            .postAsyncSafelyTask()
            .execute("CTUtils", () -> {
              ensureLocalDataStoreValue(key, cleverTapApi);
              cleverTapApi.addMultiValueForKey(key, value);
              return null;
            });
  }
}

File: AndroidSDKCore/src/main/java/com/leanplum/utils/CTUtils.kt
package com.leanplum.utils

import android.annotation.SuppressLint
import com.clevertap.android.sdk.CleverTapAPI
import com.clevertap.android.sdk.task.CTExecutorFactory
import com.clevertap.android.sdk.task.Task

object CTUtils {

  @SuppressLint("RestrictedApi")
  fun ensureLocalDataStoreValue(key: String, cleverTapApi: CleverTapAPI) {
    val value = cleverTapApi.coreState.localDataStore.getProfileProperty(key)
    if (value == null) {
      cleverTapApi.coreState.localDataStore.updateProfileFields(mapOf(key to ""))
    }
  }

  @SuppressLint("RestrictedApi")
  fun addMultiValueForKey(key: String, value: String, cleverTapApi: CleverTapAPI) {
    CTExecutorFactory
      .executors(cleverTapApi.coreState.config)
      .postAsyncSafelyTask<Task<Void>>()
      .execute("CTUtils") {
        ensureLocalDataStoreValue(key, cleverTapApi)
        cleverTapApi.addMultiValueForKey(key, value)
        null
      }
  }
}
